pkg/controller/direct/bigqueryconnection/connection_externalresource.go

```diff
@@ -3,12 +3,24 @@ package bigqueryconnection
 import (
 	"fmt"
 	"strings"
+
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/bigqueryconnection/v1alpha1"
 )
 
+type ErrNoServiceGeneratedID struct {
+}
+
+func (e ErrNoServiceGeneratedID) Error() string {
+	return fmt.Sprintf("ConfigConnector requires a GCP service-generated ID for %s. If not given, it will creata new resource",
+		krm.BigQueryConnectionConnectionGVK)
+}
+
 // The Identifier for ConfigConnector to track the BigQueryConnectionConnection resource from the GCP service.
 type BigQueryConnectionConnectionIdentity struct {
-	Parent     *parent
-	Connection string
+	Parent             *parent
+	resourceID         string
+	serviceGeneratedID string
```

Comment:
Do we actually want to distinguish `resourceID` vs. `serviceGeneratedID`?

Reply:
Good question. This mostly comes from that I think we need a dedicated `status.serviceGeneratedID` field to store the service ID for users since no write back to spec. But we shall use the service-id to replace the resourceID.

Reply:
Can we can get it from the externalRef?  I think we can, and if we can I think we should, so that we don't store the same value twice - because if we store it twice we have to worry about what happens if the values are different.

Reply:
That's true, and could be really annoying (though I don't think that should happen otherwise it is a bug because KCC controllers both fields). I think what matters is the key word `serviceGeneratedID` to acknowledge users where to find the ID. But I'm fine to remove it here.